pykafka/balancedconsumer.py

```diff
@@ -386,7 +407,10 @@ def _add_self(self):
         number of partitions.
         """
         participants = self._get_participants()
-        if len(self._topic.partitions) <= len(participants):
+        if self._consumer_id in participants:
+            return
+        if len(self._topic.partitions) < len(participants):
```

Comment:
why do you remove the equal condition? If the number of participants and the number of partitions are equal, the consumer should not register itself.